### PR TITLE
feat: improve landing page

### LINKED ISSUE #36775: The landing page should be improved
The landing page should undergo the following changes:

1. whitespace should be managed in a way that the last bullet point stays above fold.
2. add the text "As seen in:" centered and above the logos image
3. give margin/padding to the bullet points  so they are vertically aligned with the title of the page to be as same as the CTA button.

4.  Minimize text wrapping for smaller screens.

image

image

### PR DESCRIPTION
**Pull request by ahmaxed**
Closes #36775

### CODE REVIEW

File: client/src/pages/learn.js [hunk 1]
```diff
@@ -46,7 +47,10 @@ const propTypes = {
     complete: PropTypes.bool,
     errored: PropTypes.bool
   }),
+  hash: PropTypes.string,
```
Comment (ojeytonwilliams) [inline client/src/pages/learn.js]:
```suggestion
```
The landing page doesn't pass a `hash`, just `location.state`.

File: client/src/pages/learn.js [hunk 2]
```diff
@@ -68,8 +72,16 @@ const BigCallToAction = isSignedIn => {
   return '';
 };
 
+// choose between the state from landing page and hash from url.
+const hashValueSelector = (state, hash) => {
```
Comment (ojeytonwilliams) [inline client/src/pages/learn.js]:
```suggestion
const hashValueSelector = (state) => {
```

File: client/src/pages/learn.js [hunk 3]
```diff
@@ -68,8 +72,16 @@ const BigCallToAction = isSignedIn => {
...
+  if (state && state.superBlock) return dasherize(state.superBlock);
+  else if (hash) return hash.substr(1);
```
Comment (ojeytonwilliams) [inline client/src/pages/learn.js]:
```suggestion
```

File: client/src/pages/learn.js [hunk 4]
```diff
@@ -84,6 +96,8 @@ export const LearnPage = ({
     return <Loader fullScreen={true} />;
   }
 
+  const hashValue = hashValueSelector(state, hash);
```
Comment (ojeytonwilliams) [inline client/src/pages/learn.js]:
```suggestion
  const hashValue = hashValueSelector(state);
```

**Review by ojeytonwilliams — DISMISSED** [review]
I tried it out and everything seems to be working nicely.

I made some suggestions to remove references to `hash` from `learn.js` since the landing page doesn't pass it that prop (and it doesn't seem to be needed anyway).

**Comment by ahmadabdolsaheb** [comment]
@ojeytonwilliams Thanks for the review.

My apologies for the lack of explanation. 

The hash is passed from the learn page and allows for /learn#dasherized(superBlock) external navigation, so users could land on specific certificates in the map using certificates' #id, and have the map expand and scroll to the specified certificate accordingly.

**Review by raisedadead — COMMENTED** [review]
The functionality looks great, I would like to also have the scroll work on the current challenge. That is when a user is on the a challenge the `projects` link on the nav can point to the hash of the current challenge. 

That way the map would also automatically scroll to the current challenge. It would be a very subtle but nice touch IMHO.

**Comment by ojeytonwilliams** [comment]
> My apologies for the lack of explanation.

Not at all, I just didn't understand what was going on.

> The hash is passed from the learn page and allows for /learn#dasherized(superBlock) external navigation, so users could land on specific certificates in the map using certificates' #id, and have the map expand and scroll to the specified certificate accordingly.

Ah, thanks, I think I understand now!

**Comment by ahmadabdolsaheb** [comment]
> The functionality looks great, I would like to also have the scroll work on the current challenge. That is when a user is on the a challenge the `projects` link on the nav can point to the hash of the current challenge.
> 
> That way the map would also automatically scroll to the current challenge. It would be a very subtle but nice touch IMHO.

Will get it done first thing in the morning

**Comment by raisedadead** [comment]
Perfect. As soon as you have we can land this and test on the beta.

**Comment by ojeytonwilliams** [comment]
@ahmadabdolsaheb this is looking great!  The only thing I'd recommend is avoiding `componentWillMount` as it's deprecated.

I've written some code locally that makes the necessary changes.  Should I just add a commit to this so you can take a look at it?

**Comment by ahmadabdolsaheb** [comment]
@ojeytonwilliams, yes please.

**Comment by ojeytonwilliams** [comment]
I also changed the parameters, since it seemed redundant to pass in something that was already in `this.props`.

By the way, thanks for fixing my logic in `initializeExpandedState` - I'd completely forgotten to handle the case that `currentChallengeId` is not null, but isn't the id of any challenge.  Ooops.

**Comment by ahmadabdolsaheb** [comment]
@ojeytonwilliams let me know if you would like me to adjust the tests to the new changes. 
I don't want to be stepping on your feet if you are working on it.

**Comment by ojeytonwilliams** [comment]
@ahmadabdolsaheb my mistake, I'll update those now!

**Comment by ojeytonwilliams** [comment]
As well as tidying up the tests I fixed a few typos in the mocks (none of which seemed to affect anything other than the snapshots).

**Comment by ahmadabdolsaheb** [comment]
@raisedadead should we merge this for the scrolling functionality or leave it out for now?

**Comment by raisedadead** [comment]
> leave it out for now?

Please elaborate, we are shipping out in a couple days.

**Comment by ahmadabdolsaheb** [comment]
@raisedadead the requirements has changed since I started working on the PR.
I was wondering if you find any value in merging this to take advantage of the scroll implementation.
Otherwise we could close this.

**Comment by raisedadead** [comment]
> @raisedadead the requirements has changed since I started working on the PR.
> I was wondering if you find any value in merging this to take advantage of the scroll implementation.
> Otherwise we could close this.

Thanks for the clarification :) 

We will merge and land this regardless. I know it's difficult when specs change midway, but being agile means we do things in iterations. The updated specs will have to wait in the next PR.

Is this ready, with the original specs?

**Review by raisedadead — APPROVED** [review]

**Pull request merged**